src/features/environment.js

```diff
@@ -167,6 +176,16 @@ export const EnvironmentMixin = (ModelViewerElement) => {
       this[$needsRender]();
     }
 
+    [$updateShadow]() {
+      const {shadowStrength} = this;
+      const shadowStrengthIsNumber =
+          typeof shadowStrength === 'number' && !Number.isNaN(shadowStrength);
+
+      this[$scene].shadow.material.opacity = BASE_SHADOW_OPACITY *
+          (shadowStrengthIsNumber ? shadowStrength : DEFAULT_SHADOW_STRENGTH);
```

Comment:
I think I prefer moving some of the internal logic into the component, e.g. `this[$scene].shadow.setIntensity(...)` and let the component handle the base shadow opacity, similar to `model.applyEnvironmentMap(...)` in this module, and leave the feature focused on plumbing the different configurations/components, and let the components handle the details. This idea sort of fell a part a bit with the removal of the Skysphere, since `scene.background` is being manipulated directly here, and we don't have an abstraction/three-component for it otherwise.

Reply:
I don't think it'll matter much, but wonder if there's anything we need to do with the default case (e.g. rendering an opacity=0 mesh) -- I think it'll be skipped in the renderer somewhere, and the only potential overhead is the initial creation of the shadow. Even if we're rendering an opacity=0 shadow, it's a fake shadow anyway, so a negligible cost (outside of possibly the LUT on mobile??)

Reply:
Will fix